3DCalendar/CalendarViewController.swift

```diff
@@ -0,0 +1,197 @@
+        return viewController
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        collectionView.dataSource = self
+        collectionView.delegate = self
+
+        initializeView()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
+    
+    func initializeView() {
+        currentMonthIndex = inputIndex
+        currentYear = Calendar.current.component(.year, from: Date())
+        todaysDate = 1
+        firstWeekDayOfMonth = getFirstWeekDay()
+        
+        //leap years
+        if currentMonthIndex == 2 && currentYear % 4 == 0 {
+            numDaysPerMonth[currentMonthIndex-1] = 29
+        }
+        
+        presentMonthIndex = currentMonthIndex
+        presentYear = currentYear
+        
+        setUpViews()
+        
+        collectionView.collectionViewLayout = UICollectionViewFlowLayout()
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        let colors: [UIColor] = [.yellow, .cyan, greeny, lightPink, aqua]
+        let rando = colors.randomElement()
+        self.view.backgroundColor = rando
+        collectionView.backgroundColor = rando
+        collectionView.allowsMultipleSelection = false
+        
+        collectionView.register(DateCell.self, forCellWithReuseIdentifier: "Cell")
+        let date = Date()
+        monthLabel.text = "\(date.month(monthIndex: currentMonthIndex-1))"
+    }
+   
+    func getFirstWeekDay() -> Int {
+        let day = ("\(currentYear)-\(currentMonthIndex)-01".date?.firstDayOfTheMonth.weekday)!
+        //return day == 7 ? 1 : day
+        return day
+    }
+    
+    func setUpViews() {
+//        self.view.addSubview(myCollectionView)
+        self.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+//        myCollectionView.topAnchor.constraint(equalTo: self.view.topAnchor).isActive = true
+//        myCollectionView.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
+//        myCollectionView.rightAnchor.constraint(equalTo: self.view.rightAnchor).isActive = true
+//        myCollectionView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor).isActive = true
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == collectionView {
+            return numDaysPerMonth[currentMonthIndex-1] + firstWeekDayOfMonth - 1
+        } else {
+            return weekdayData.count
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! DateCell
+        cell.frame.size.width = 25.0
```

Comment:
Even tho this is small we may still want to put it in a nib. It'll help get rid of a lot of the sizing, color, and interaction logic you have in this function.